primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -794,4 +794,27 @@ public static Object convertToType(Object value, Class<?> valueType, Logger logg
             return value;
         }
     }
+
+    public static <T> T findComponent(UIComponent root, Class<T> childType) {
```

Comment:
just use ComponentTraversalUtils#first

Reply:
I'm not too sure yet about using ComponentTraversalUtils (as it traverses the children's children) By introducing `ComponentUtils#findComponent` (which only visits self or children and take account rendered attr) i was planning to reuse that method in various places where sometimes we need to check the presence of a component (for instance: celleditor, rowexpansion etc.)

Reply:
then we need a better name for it
findComponent is something different already in JSF